File: routes/reservations.js
const express = require('express');
const router = express.Router();
const logger = require('../utils/logger');

// Get today's reservations for a user
router.get('/user/:userId/today', async (req, res) => {
  try {
    const { userId } = req.params;
    const reservationService = req.app.locals.services.reservation;
    
    if (!reservationService) {
      return res.status(503).json({
        success: false,
        error: 'Reservation service not available'
      });
    }

    const reservations = await reservationService.getTodayReservations(userId);
    
    res.json({
      success: true,
      data: reservations
    });
  } catch (error) {
    logger.error('Get today reservations error:', error);
    
    res.status(500).json({
      success: false,
      error: 'Failed to get reservations'
    });
  }
});

// Confirm reservation (mark as served)
router.post('/:id/confirm', async (req, res) => {
  try {
    const { id } = req.params;
    const reservationService = req.app.locals.services.reservation;
    const purchaseService = req.app.locals.services.purchase;
    
    if (!reservationService) {
      return res.status(503).json({
        success: false,
        error: 'Reservation service not available'
      });
    }

    const result = await reservationService.confirmReservation(id, purchaseService);
    
    logger.logUserAction('Reservation Confirmed via RFID', result.reservation.studentId, {
      reservationId: id,
      cafeteria: process.env.CAFETERIA_NAME
    });

    res.json(result);
  } catch (error) {
    logger.error('Confirm reservation error:', error);
    
    res.status(400).json({
      success: false,
      error: error.message || 'Failed to confirm reservation'
    });
  }
});

// Get reservation details
router.get('/:id', async (req, res) => {
  try {
    const { id } = req.params;
    const reservationService = req.app.locals.services.reservation;
    
    if (!reservationService) {
      return res.status(503).json({
        success: false,
        error: 'Reservation service not available'
      });
    }

    const reservation = await reservationService.getReservationById(id);
    
    if (!reservation) {
      return res.status(404).json({
        success: false,
        error: 'Reservation not found'
      });
    }

    res.json({
      success: true,
      data: reservation
    });
  } catch (error) {
    logger.error('Get reservation error:', error);
    
    res.status(500).json({
      success: false,
      error: 'Failed to get reservation'
    });
  }
});

module.exports = router;

File: flaskrfid/routes/reservations.py
from flask import Blueprint, request, jsonify, current_app
import structlog

logger = structlog.get_logger()
reservations_bp = Blueprint('reservations', __name__)

@reservations_bp.route('/user/<user_id>/today', methods=['GET'])
def get_today_reservations(user_id):
    try:
        reservation_service = current_app.reservation_service
        
        if not reservation_service:
            return jsonify({
                'success': False,
                'error': 'Reservation service not available'
            }), 503
        
        reservations = reservation_service.get_today_reservations(user_id)
        
        return jsonify({
            'success': True,
            'data': reservations
        })
    except Exception as e:
        logger.error("Get today reservations error", error=str(e))
        
        return jsonify({
            'success': False,
            'error': 'Failed to get reservations'
        }), 500

@reservations_bp.route('/<reservation_id>/confirm', methods=['POST'])
def confirm_reservation(reservation_id):
    try:
        reservation_service = current_app.reservation_service
        purchase_service = current_app.purchase_service
        
        if not reservation_service:
            return jsonify({
                'success': False,
                'error': 'Reservation service not available'
            }), 503
        
        result = reservation_service.confirm_reservation(reservation_id, purchase_service)
        
        logger.info("Reservation Confirmed via RFID", 
                   student_id=result['reservation']['student_id'],
                   reservation_id=reservation_id,
                   cafeteria=current_app.config.get('CAFETERIA_NAME'))
        
        return jsonify(result)
    except Exception as e:
        logger.error("Confirm reservation error", error=str(e))
        
        return jsonify({
            'success': False,
            'error': str(e) or 'Failed to confirm reservation'
        }), 400

@reservations_bp.route('/<reservation_id>', methods=['GET'])
def get_reservation(reservation_id):
    try:
        reservation_service = current_app.reservation_service
        
        if not reservation_service:
            return jsonify({
                'success': False,
                'error': 'Reservation service not available'
            }), 503
        
        reservation = reservation_service.get_reservation_by_id(reservation_id)
        
        if not reservation:
            return jsonify({
                'success': False,
                'error': 'Reservation not found'
            }), 404
        
        return jsonify({
            'success': True,
            'data': reservation
        })
    except Exception as e:
        logger.error("Get reservation error", error=str(e))
        
        return jsonify({
            'success': False,
            'error': 'Failed to get reservation'
        }), 500